python/paddle/v2/fluid/io.py

```diff
@@ -144,42 +182,64 @@ def load_vars(executor, dirname, main_program=None, vars=None, predicate=None):
         load_vars(
             executor,
             dirname=dirname,
-            vars=filter(predicate, main_program.list_vars()))
+            vars=filter(predicate, main_program.list_vars()),
+            load_file_name=load_file_name)
     else:
         load_prog = Program()
         load_block = load_prog.global_block()
-        for each_var in vars:
-            assert isinstance(each_var, Variable)
-            new_var = _clone_var_in_block_(load_block, each_var)
+
+        if load_file_name is None:
+            for each_var in vars:
+                assert isinstance(each_var, Variable)
+                new_var = _clone_var_in_block_(load_block, each_var)
+                load_block.append_op(
+                    type='load',
+                    inputs={},
+                    outputs={'Out': [new_var]},
+                    attrs={'file_path': os.path.join(dirname, new_var.name)})
+        else:
+            load_var_map = {}
+            for each_var in vars:
+                assert isinstance(each_var, Variable)
+                new_var = _clone_var_in_block_(load_block, each_var)
```

Comment:
Move the common codes `line 202 - 204` out of the `if` statement?

Reply:
Done